src/main/java/com/nyanbot/DukeTasks/Task.java

```diff
@@ -0,0 +1,316 @@
+                    // add the year and time to the final date
+                    temp = addYear(temp);
+                    temp = addTime(temp);
+                    if (temp.isBefore(LocalDateTime.now())) {
+                        temp = temp.plusYears(1);
+                    }
+                }
+                // gc: the given date is before today's date even after parsing
+                if (temp.isBefore(LocalDateTime.now())) {
+                    this.setInvalid();
+                    DukeExceptionHandlers.handleNoDate(error);
+                }
+                return temp;
+            } catch (ParseException ignored) {
+            }
+        }
+        this.setInvalid();
+        DukeExceptionHandlers.handleNoDate(error);
+        return null;
+    }
+
+    /**
+     * Checks if the given String is numeric or not.
+     *
+     * @author Tan Kerway
+     * @param str the string to check
+     * @return true if the string is numeric, false otherwise
+     */
+    private boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException ignored) {}
+        return false;
+    }
+
+    /**
+     * Adds the year to the date object.
+     *
+     * @author Tan Kerway
+     * @param time the LocalDateTime object that represents start dates, or deadlines
+     * @return the modified LocalDateTime object
+     */
+    private LocalDateTime addYear(LocalDateTime time) {
+        LocalDateTime now = LocalDateTime.now();
+        int yearsElapsed = now.getYear() - DEFAULT_DATE.getYear();
+        // update the year
+        return time.plusYears(yearsElapsed);
+    }
+
+    /**
+     * Adds the day to the date object.
+     *
+     * @author Tan Kerway
+     * @param time the LocalDateTime object that represents start dates, or deadlines
+     * @return the modified LocalDateTime object
+     */
+    private LocalDateTime addDay(LocalDateTime time) {
+        LocalDateTime now = LocalDateTime.now();
+        int defaultDay = DEFAULT_DATE.getDayOfWeek().getValue();
+        int timeDay = time.getDayOfWeek().getValue();
+        int daysElapsed = timeDay >= defaultDay ? timeDay - defaultDay : 7 - defaultDay + timeDay;
+        // update the current date by the number of days
+        LocalDateTime res = now.plusDays(daysElapsed);
+        if (time.getHour() == 0 && time.getMinute() == 0) {
+            return addTime(res);
+        }
+        return res.withHour(time.getHour()).withMinute(time.getMinute());
+    }
+
+    /**
+     * Updated the time fields of the LocalDateTime object.
+     *
+     * @author Tan Kerway
+     * @param time the given LocalDateTime object
+     * @return a new LocalDateTime object, with the same parameters as time, but with hour field
+     *         set to 23 and minute field set to 59
+     */
+    private LocalDateTime addTime(LocalDateTime time) {
+        return time.withHour(23).withMinute(59);
```

Comment:
For the integers such as 23 and 59, maybe it would be better to create a private static final class for them? since what they represent is not apparent at first glance.